src/Domain/FunctionsOfTime/SettleToConstantQuaternion.cpp

```diff
@@ -0,0 +1,136 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/FunctionsOfTime/SettleToConstantQuaternion.hpp"
+
+#include <cmath>
+#include <limits>
+#include <memory>
+
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeArray.hpp"
+
+namespace {
+template <size_t MaxDerivReturned>
+std::array<double, MaxDerivReturned + 1> quat_norm_and_derivs(
+    const std::array<DataVector, MaxDerivReturned + 1>&
+        unnormalized_func_and_derivs) {
+  std::array<double, MaxDerivReturned + 1> result =
+      make_array<MaxDerivReturned + 1>(0.0);
+  gsl::at(result, 0) =
+      sqrt(sum(square(gsl::at(unnormalized_func_and_derivs, 0))));
+  if constexpr (MaxDerivReturned > 0) {
+    gsl::at(result, 1) = sum(gsl::at(unnormalized_func_and_derivs, 0) *
```

Comment:
[Optional] For `std::array`s, you only have to use `gsl::at()` when accessing with a non-const variable (like `i`). If you're accessing with a constant expression (i.e. a number like `0` or `1`), then you can use `[]` to access.

Reply:
Unless others insist, I'd rather stick with gsl::at(), but good to know for the future

Reply:
Yeah that's alright.